pkg/connector/collection.go

```diff
@@ -43,12 +45,12 @@ func (o *collectionBuilder) List(ctx context.Context, parentResourceID *v2.Resou
 	}
 
 	if parentResourceID.ResourceType != databaseResourceType.Id {
-		return nil, "", nil, fmt.Errorf("invalid parent resource type: %s", parentResourceID.ResourceType)
+		return nil, "", nil, uhttp.WrapErrors(codes.InvalidArgument, "mongo-db-connector: invalid parent resource type", fmt.Errorf("expected %s, got %s", databaseResourceType.Id, parentResourceID.ResourceType))
```

Comment:
We should not use codes.InvalidArgument here as that error is for when the customer did not configure the connector correctly. The point of the GRPC status wrapping is to reduce the amount of errors we see that can be handled by customers.